### PR TITLE
make it run the tests, add initial service impl

### PR DESCRIPTION
**Pull request by MysterionRise**

### CODE REVIEW

File: src/test/kotlin/com/github/galleyltd/boost/service/AnalysisServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.junit.jupiter.api.Test
+
+class AnalysisServiceTest {
+
+    private val koinContainer = KoinContainer().also { it.init() }
+    private val openDotaApiClient = koinContainer.openDotaApiClient
+    private val service = SimpleAnalysisService()
+
+    private suspend fun requestTestMatches(ids: List<String>): List<MatchData> {
```
Comment (IgorPerikov) [inline src/test/kotlin/com/github/galleyltd/boost/service/AnalysisServiceTest.kt]: I'd suggest you you to move `runBlocking` into helper method

File: src/main/kotlin/com/github/galleyltd/boost/service/AnalysisSerivce.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+class SimpleAnalysisService : AnalysisSerivce {
+
+    private val WINDOW_SIZE = 5
+    private val STEP = 1
+    private val DIFFERENCE = 0.2
+
+    override fun accountFeedback(matches: List<MatchData>, accountId: Int): AccountFeedback {
+        println()
+        val xpmData = matches.map { it.players.first { it -> it.accountId == accountId }.xpPerMin }
+            .map { it -> it }
```
Comment (IgorPerikov) [inline src/main/kotlin/com/github/galleyltd/boost/service/AnalysisSerivce.kt]: I guess it's redundant

File: src/main/kotlin/com/github/galleyltd/boost/service/AnalysisSerivce.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package com.github.galleyltd.boost.service
+
+import com.github.galleyltd.boost.opendota.dto.MatchData
+import kotlin.math.abs
+
+interface AnalysisSerivce {
+
+    fun accountFeedback(matches: List<MatchData>, accountId: Int): AccountFeedback
+    fun relativeDifference(v1: Double, v2: Double): Double {
```
Comment (IgorPerikov) [inline src/main/kotlin/com/github/galleyltd/boost/service/AnalysisSerivce.kt]: it's not a part of a service contract I guess? we can move it out of the service scope, maybe even private extension function :thinking:

File: src/main/kotlin/com/github/galleyltd/boost/service/AnalysisSerivce.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        val movingAverageKDA =
+            kdaData.windowed(WINDOW_SIZE, STEP) { it.average() }.windowed(WINDOW_SIZE, STEP) { it.average() }
+
+        return AccountFeedback(
+            accountId,
+            relativeDifference(movingAverageXPM.max()!!, movingAverageXPM.min()!!) > DIFFERENCE,
+            relativeDifference(movingAverageGPM.max()!!, movingAverageGPM.min()!!) > DIFFERENCE,
+            relativeDifference(movingAverageKPM.max()!!, movingAverageKPM.min()!!) > DIFFERENCE,
+            relativeDifference(movingAverageHD.max()!!, movingAverageHD.min()!!) > DIFFERENCE,
+            relativeDifference(movingAverageKDA.max()!!, movingAverageKDA.min()!!) > DIFFERENCE
```
[the remark this answers is not in the document]
Comment (MysterionRise) [reply src/main/kotlin/com/github/galleyltd/boost/service/AnalysisSerivce.kt]: I'm unaware of something simple and clean enough that will be better than this

**Pull request merged**